lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -245,92 +220,126 @@ def eci_to_result_indicator(eci)
 
-        response = parse(ssl_post(url, post_data(xml), 'Content-Type' => 'application/x-www-form-urlencoded;charset=UTF-8'))
+        response = api_request(url + endpoint, params(request, action), 'Content-Type' => 'application/json;charset=UTF-8')
+        success = success_from(response)
 
         Response.new(
-          success_from(response),
-          message_from(response),
+          success,
+          message_from(response, success),
           response,
           authorization: authorization_from(response),
           test: test?,
-          error_code: error_code_from(response)
+          error_code: error_code_from(response, success)
         )
       end
 
       # Private: Decide success from servers response
       def success_from(response)
-        response[:status] == :success || response[:status] == :new
+        response['result'] === 'completed'
       end
 
       # Private: Get message from servers response
-      def message_from(response)
-        response[:message]
+      def message_from(response, success)
+        success ? 'Transaction approved' : response.fetch('message', 'No error details')
       end
 
       # Private: Get error code from servers response
-      def error_code_from(response)
-        success_from(response) ? nil : STANDARD_ERROR_CODE[:card_declined]
+      def error_code_from(response, success)
+        success ? nil : STANDARD_ERROR_CODE[:card_declined]
       end
 
       # Private: Get authorization code from servers response
       def authorization_from(response)
-        response[:unique_id]
+        response['monei_order_id']
       end
 
       # Private: Encode POST parameters
-      def post_data(xml)
-        "load=#{CGI.escape(xml)}"
+      def post_data(params)
+        params.clone.to_json
       end
 
-      # Private: Translate Monei status code to native ruby symbols
-      def translate_status_code(code)
-        {
-          '00' => :success,
-          '40' => :neutral,
-          '59' => :waiting_bank,
-          '60' => :rejected_bank,
-          '64' => :waiting_risk,
-          '65' => :rejected_risk,
-          '70' => :rejected_validation,
-          '80' => :waiting,
-          '90' => :new
-        }[code]
+      # Private: generate request params depending on action
+      def params(request, action)
+        if action == :purchase || action == :authorize
+          request = {
+            'charge': request,
+            'context': {
+              ip: options[:ip] || '0.0.0.0',
+              userAgent: options[:user_agent] || 'ActiveMerchant UA'
+            }
+          }
+        end
+        request
       end
 
       # Private: Translate AM operations to Monei operations codes
       def tanslate_payment_code(action)
         {
-          purchase: 'CC.DB',
-          authorize: 'CC.PA',
-          capture: 'CC.CP',
-          refund: 'CC.RF',
-          void: 'CC.RV'
+          purchase: 'sale',
```

Comment:
This private method name has a typo in it. `tanslate_payment_code` => `translate_payment_code`.

Perhaps this breaking-changes PR is the time to clean that up?

Reply:
Oh nice catch! I'll do it, thanks